### ISSUE TITLE
Can't update to 12.1 Version

### ISSUE THREAD

**Issue by phatair** [vmware/open-vm-tools#609]
### Describe the bug

I know its not a bug but i dont know how to ask this question.
We run Ubuntu 20.04 and if we do a apt-get update and apt-get upgrade the open-vm-tools are not updated.
I got the message that we run the newest version.
This is 11.3.0.29534 (build-18090558)

But the newest version should be 12.1

How can i update the open-vm-tools 11.3 to 12.1 on Ubuntu 20.04?
Thanks so much for your help.
Best regards

### Reproduction steps

```bash
1. 11.3.0.29534 (build-18090558) installed on ubuntu 20.04
2. run apt-get upgrade or apt-get install open-vm-tools
3. no update available
```

### Expected behavior

Version should be updated to 12.2

**Comment by johnwvmw** [comment]
You probably will not be able to get open-vm-tools-12.1.0 for Ubuntu 22.04.  Available open-vm-tools versions available for a Linux release varies between  Linux vendors .

Ubuntu will make OVT 12.1.0 available on 22.10.

The presence of open-vm-tools 11.3.0 on your system suggests that you may not have availed yourself to the updated packages and security fixes provided by Ubuntu.    Ubuntu provides open-vm-tools 11.3.5 for UB 22.04.  

apt list --upgradeable | grep open-vm-tools shows that there is an OVT 11.3.5 package updates in the jammy-security repository.

apt update
apt upgrade open-vm-tools-desktop open-vm-tools
   --- updates numerous packages.....
apt list --installed   | grep open-vm-tools
.... shows 
open-vm-tools-desktop/jammy-updates,jammy-security,now 2:11.3.5-1ubuntu4.1 amd64 [installed]
open-vm-tools/jammy-updates,jammy-security,now 2:11.3.5-1ubuntu4.1 amd64 [installed,automatic]

**Comment by johnwvmw** [comment]
I believe that the jammy-security open-vm-tools packages are addressing CVE-2022-31676

**Comment by phatair** [comment]
Thanks @johnwvmw 
Your last post means, that this update will be published by ubuntu, right?

**Comment by johnwvmw** [comment]
That is my expectation.   I was able to upgrade OVT 11.0.,5 packages on my test UB 20.04.3 system yesterday.

Please run `apt list --upgradeable | grep open-vm-tools` to check for availability.

**Comment by phatair** [comment]
Hi Johnwvmw
if i run `apt list --upgradeable | grep open-vm-tools` there are no results
If i run 'apt list| grep open-vm-tools' i get this
```

> open-vm-tools-desktop/focal-updates,focal-security 2:11.3.0-2ubuntu0~ubuntu20.04.3 amd64
> open-vm-tools-dev/focal-updates,focal-security 2:11.3.0-2ubuntu0~ubuntu20.04.3 amd64
> open-vm-tools-sdmp/focal-updates,focal-security 2:11.3.0-2ubuntu0~ubuntu20.04.3 amd64
> open-vm-tools/focal-updates,focal-security,now 2:11.3.0-2ubuntu0~ubuntu20.04.3 amd64 [installed]
```

For me, it looks like that the new update is not found :(

**Comment by johnwvmw** [comment]
Those are the updated (fixed) 11.3.0 open-vm-tools packages.   Note the **_focal-security_**.

Prior to the public VMSA announcement, Linux vendors are notified of the pending announcement and provided with patches that can be applied to all affected versions.

From the fact the the "--upgradeable" does not show any OVT, and "apt list" does show the security updates, it appears that you previously updated the 11.3.0 open-vm-tools after Ubuntu made the fixes available.

To confirm:
* Run "apt list --installed | grep open-vm-tools"
* To check when the open-vm-tools packages where installed, run
       grep open-vm-tools /var/log/dpkg*
       
 Note:  Previous versions of /var/log/dpkg.log file are there, but compressed.

**Comment by phatair** [comment]
Thanks johnwvmw,

the open-vm-tools update was installed on 2022-08-25

If i run the command apt list --installed | grep open-vm-tools i get this result
/focal-updates,focal-security,now 2:11.3.0-2ubuntu0~ubuntu20.04.3 amd64 [installed]

so we have installed the newest open-vm-tools version with the fix for the CVE-2022-31676, right?

**Comment by johnwvmw** [comment]
@phatair    Correct.

Closing this issue.  Please reopen if there are further questions.

**Comment by Muy69** [comment]
We have now 2024 and there is still no VMWARE 12.x Version for ubuntu20.04.6 . 

Currently I get flagged that 11.3 is a critical VMWARE VMWARETOOLS Version , is it planed that a open-vm-tools Version will be avaible for 20.04 as it is with 22.04 ?

**Comment by Muy69** [comment]
@phatair I would like to reopen the Request or should I file a New One ?

**Comment by johnwvmw** [comment]
@phatair

As CVE fixes are provides in the open-vm-tools releases, we provide a complete set of CVE fixes for previous open-vm-tools releases.  Each Linux vendor decides how they will address the security fixes - based on where a Linux release is in its lifecycle.  Early in the lifecycle, a newer version of OVT may be issued.   Later in the lifecycle, updated pacjage will be made available with the security fixes.   

Ubuntu is quite responsive with CVE security fixes when announced for their LTS releases.

I see that  open-vm-tools packages "2:11.3.0-2ubuntu0~ubuntu20.04.7 [security]: amd64" is available.   Note the "20.04.7" in the package version.   This is 4 package updates to OVT 11.3.0 since your last mentioned update in Aug. 2022.

Just install the latest security updates and you should be good.

**Comment by Muy69** [comment]
@phatair @johnwvmw 
So Development of Feature are already  frozen and only CVE are fixed as needed  , for a Ubuntu 20.04 LTS ? 

But 20.04.07 is still a Product under LTS Support until 2025 , what is the intention of skipping on current Funktion which the Jammy Jellyfish Release has ?  

I dont understand the Logic to that Decision , it was already a valid Question in **2022** and it is also in **2024** . 

The 11.3 VMWARE Tools were release in 2021.

**Comment by johnwvmw** [comment]
You were not asking about open-vm-tools on 22.04 but asking about the LTS 20.04 Ubuntu release.

Because 20.04 is LTS, Ubuntu continues to support 20.04 and has fixed every announced CVE in the 11.3.x that they provided for the 20.04 release.

Upgrading to a new verison of open-vm-tools with new features can involve required updates to various open source packages required by OVT.   Most open source projects do not support multiple versions of their runtime on the same system at the same time.  For example, "fuse" and "fuse3" are at issue in this case.   Ubuntu switched from fuse (fuse 2) to fuse3 at Ubuntu 21.04, I believe. 

Rev'ing the runtime of an open source project has a ripple effect in that all other projects using that rev'ed runtime must be rebuilt and retested.

VMware Inc .and now VMware on Broadcom do not produce nor dictate the version of open-vm-tools packages provided by the Linux vendors.  We provide the open-vm-tools source to Linux vendors and security fix patches for all recent version of open-vm-tools.  The Linux vendors decide what versions of OVT will be available on their Linux releases.

My statement in Aug. 2022 that OVT 12.1.0 may not be available for 22.04 was what we were told at that time as tacked through various LaunchPad PRs.  OVT 12.1.0 was only targeted for the 22.10 release.   It has only been within the last 3 months, that Ubuntu development engineers and their security team have agreed, after much discussion, to provide OVT 12.3.5 for Ubuntu 22.04 LTS.

The decision not to use a later version of open-vm-tools on 20.04 (focal) is solely Ubuntu's as seen in the referenced LauchPad incident.

**Comment by Muy69** [comment]
Yes Im asking about Ubuntu 20.04 LTS  as @phatair startet the Thread  in 2022. ( But you pointed out already why Ubuntu  22.04 LTS  there were different decisions made ) 

Thanks for the insight, so I understand that Unbutu/Canocial  decided to only support the 22.04 LTS  and later on Distribution with current OVT  **only** . 

So when a VMWARE Tools 13.x will arrive it will probably also not likley seen by only  **not** Jammy Version as the Version Number and Funktion  is frozen to the current 12.x . 

That forces Ubuntu LTS Version  on VMWARE Enviroments to evaluat early the Move to  the "newest"  Ubuntu  LTS Installation,  to leverage new OVT Funktion exclusivley added to new Distrubutions.

Given that info Windows VM have an advantage over Linux VM in terms of  VMWARE Tools , they receive not only CVE Updates but also  Funktions too.  

Therefor [Salt-Minion](https://docs.vmware.com/de/VMware-Tools/12.3.0/com.vmware.vsphere.vmwaretools.doc/GUID-373CD922-AF80-4B76-B19B-17F83B8B0972.html)  is implimented in an EOL Version / Windows 2012   Vmware Version / Development  ( VMWARE tools ) but not on a still supported Version Ubuntu  20.04 LTS ( 2025 )   as **Ubuntu / Canonical** decided against,  even the OVT Code is avaible  ? 

So that is the reason behind the prompt closing of OVT Version 12.3.5 Request  for the Ubuntu Version 20.04 LTS. 
https://github.com/vmware/open-vm-tools/issues/709

I was not aware that such behaviour exist,  as in the Changelog the Support Version "ubuntu" is only understandble in context of your explanation and in reflection  with the  LaunchPad, I find it irritating , why Windows and Linux VMWARE Tools have such a different Approach. 

Thanks again for your insight , im not happy , but at least we know now what the Gamerules are for OVT in the Ubuntu Scope.
